client/app/models/stack_application.coffee

```diff
@@ -10,53 +10,42 @@ module.exports = class StackApplication extends Backbone.Model
         return "#{base}byid/#{@get('id')}" if @get('id')
         return base
 
-    # use same events as backbone to enable socket-listener
-    prepareCallbacks: (callbacks, presuccess, preerror) ->
-        {success, error} = callbacks or {}
-        presuccess ?= (data) => @set data.app
-        @trigger 'request', @, null, callbacks
-        callbacks.success = (data) =>
-            presuccess data if presuccess
-            @trigger 'sync', @, null, callbacks
-            success data    if success
-        callbacks.error = (jqXHR) =>
-            preerror jqXHR if preerror
-            @trigger 'error', @, jqXHR, {}
-            error jqXHR     if error
-
-
-    waitReboot: (step, total_step, callbacks) ->
-        {success, error} = callbacks or {}
-        client.get "api/applications/stack",
-            success: =>
-                if step is total_step
-                    if success?
-                        success 'ok'
-                    else
-                        callbacks() if callbacks
-                else
-                    if step is 1
-                        step += step
-                    setTimeout () =>
-                        @waitReboot step, total_step, callbacks
-                    , 500
-            error: =>
-                setTimeout () =>
-                    if step is 0 or step is 2
-                        step = step + 1
-                    @waitReboot step, total_step, callbacks
-                , 500
-
-    updateStack: (callbacks) ->
-        client.put "/api/applications/update/stack", {},
-            success: =>
-                @waitReboot 0, 2, callbacks
-            error: =>
-                @waitReboot 0, 2, callbacks
-
-    rebootStack: (callbacks) ->
-        client.put "/api/applications/reboot/stack", {},
-            success: =>
-                @waitReboot 0, 1, callbacks
-            error: =>
-                @waitReboot 0, 1, callbacks
\ No newline at end of file
+
+    # Send a series of requests to the server to ensure that it is up.
+    # The callback is fired only when three requests has been successfully
+    # responded.
+    waitServerIsUp: (step, totalStep, callback) ->
```

Comment:
It looks weird to have to call this method with `step` as first parameter. What about this?

```coffee
waitServerIsUp: (remainingSteps, callback) ->
    request.get "api/applications/stack", (err) ->
        remainingSteps-- unless err
        if remainingSteps is 0
            callback()
        else
            setTimeout =>
                @waitServerIsUp remainingSteps, callback
            , 1000
```